### PR TITLE
[improve](table function) opt explode/explode_map/explode_json table function

### PR DESCRIPTION
**Pull request by zhangstar333**
before get_value, it's will insert one row into column once, 
now could insert many result into column once, could reduce some virtual function call

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by BiteTheDDDDt — DISMISSED** [review]

File: be/src/vec/exprs/table_function/vexplode_json_array.h [hunk 1]
```diff
@@ -19,85 +19,231 @@
...
+    }
+    std::vector<UInt8> _values_null_flag;
+};
 
-    // The number parsed from json array
-    // the `_backup` saved the real number entity.
-    std::vector<void*> _data;
-    std::vector<StringRef> _data_string;
+struct ParsedDataInt : public ParsedData<int64_t> {
+    static auto constexpr max_value = std::numeric_limits<int64_t>::max(); //9223372036854775807
```
Comment (HappenLee) [inline be/src/vec/exprs/table_function/vexplode_json_array.h:53]: MAX_VALUE constexpr

File: be/src/vec/exprs/table_function/vexplode_json_array.h [hunk 2]
```diff
@@ -19,85 +19,231 @@
...
+    ParsedData() = default;
+    virtual ~ParsedData() = default;
+    virtual void reset() = 0;
+    virtual int set_output(rapidjson::Document& document, int value_size) = 0;
+    virtual void insert_result_from_parsed_data(MutableColumnPtr& column, int max_step,
+                                                int64_t cur_offset) = 0;
+    const char* get_null_flag_address(int cur_offset) {
+        return reinterpret_cast<const char*>(_values_null_flag.data() + cur_offset);
+    }
+    std::vector<UInt8> _values_null_flag;
```
Comment (HappenLee) [inline be/src/vec/exprs/table_function/vexplode_json_array.h:49]: std::vector<T> _backup

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**